### PR TITLE
[DOCS] Adds missing TLS settings for auditing

### PR DESCRIPTION
**Pull request by lcawl**
Fixes https://github.com/elastic/elasticsearch/issues/29692

### CODE REVIEW

**Review by jaymode — APPROVED** [review]
LGTM

**Comment by lcawl** [comment]
Thanks, @jaymode.  @albertzaharovits has some additional information, and I've made a few more edits too.  For now, I've put a preview of the changes here:
 
https://doc-review-63bf6.firebaseapp.com/es2/auditing-settings.html#remote-audit-settings

https://doc-review-63bf6.firebaseapp.com/es2/security-settings.html#auditing-tls-ssl-settings

https://doc-review-63bf6.firebaseapp.com/x/forwarding-audit-logfiles.html

I've split out that last page because think it might be a good idea to add a list of steps (similar to the instructions in https://www.elastic.co/guide/en/elasticsearch/reference/master/configuring-tls.html). Those details might come in a subsequent PR.

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 1]
```diff
@@ -1,24 +1,99 @@
...
+xpack.security.audit.index.client.cluster.name: logging-prod <2>
+xpack.security.audit.index.client.xpack.security.user: myuser:mypassword <3>
+--------------------------------------------------
+<1> A list of hosts in the remote cluster. If you are not using the default 
+value for the `transport.tcp.port` setting on the remote cluster, you must 
+specify the appropriate port number (prefixed by a colon) after each host. 
+<2> The remote cluster name.
+<3> A valid user and password, which must have authority to create the 
+`.security-audit` index on the remote cluster. 
+//TBD: Is there a secure version of this setting, so that password is hidden in the keystore?
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: No, there is no such setting.

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 2]
```diff
@@ -1,24 +1,99 @@
...
+--
+
+. If the remote cluster has Transport Layer Security (TLS/SSL) enabled, you 
+must specify extra security settings. 
 
-You can pass additional settings to the remote client by specifying them in the
-`xpack.security.audit.index.client` namespace. For example, to allow the remote
-client to discover all of the nodes in the remote cluster you can specify the
-`client.transport.sniff` setting:
+.. {ref}/configuring-tls.html#node-certificates[Generate a node certificate].
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: I think "Generate a node certificate" is ambiguous.
The certificate should be generated on the remote cluster indeed similar to generating ones for a node in that remote cluster, but used on the client on the local cluster node. Don't know how to phrase this lightly...

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 3]
```diff
@@ -1,24 +1,99 @@
...
+*** If the certificate is in PEM format, use the following commands:
++
+--
+[source,shell]
+-----------------------------------------------------------
+bin/elasticsearch-keystore add xpack.security.audit.index.client.xpack.ssl.secure_key_passphrase
+-----------------------------------------------------------
+--
+
+//TBD: Is this required?: . Restart {es}.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Yes, restart is required.

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 4]
```diff
@@ -1,24 +1,99 @@
...
+--
 [source,yaml]
-----------------------------
-xpack.security.audit.index.client.transport.sniff: true
-----------------------------
+-----------------------------------------------------------
+xpack.security.audit.index.client.xpack.security.transport.ssl.enabled: true
+xpack.security.audit.index.client.xpack.ssl.keystore.path: certs/elastic-certificates.p12 
+xpack.security.audit.index.client.xpack.ssl.truststore.path: certs/elastic-certificates.p12
+-----------------------------------------------------------
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Maybe these names here suggest that this is a certificate for a client.

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 5]
```diff
@@ -1,24 +1,99 @@
...
+`elasticsearch.yml` file:
++
+--
+[source, yaml]
+--------------------------------------------------
+xpack.security.audit.index.client.xpack.security.transport.ssl.enabled: true
+xpack.security.audit.index.client.xpack.ssl.key: /home/es/config/node01.key 
+xpack.security.audit.index.client.xpack.ssl.certificate: /home/es/config/node01.crt 
+xpack.security.audit.index.client.xpack.ssl.certificate_authorities: [ "/home/es/config/ca.crt" ] 
+--------------------------------------------------
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Idem comment - 1.

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc [hunk 6]
```diff
@@ -1,24 +1,99 @@
...
+//TBD: Is there a secure version of this setting, so that password is hidden in the keystore?
 
 For more information about these settings, see
-{ref}/auditing-settings.html#remote-audit-settings[Remote Audit Log Indexing Configuration Settings].
+{ref}/auditing-settings.html#remote-audit-settings[Remote audit log indexing configuration settings].
+
+--
+
+. If the remote cluster has Transport Layer Security (TLS/SSL) enabled, you 
+must specify extra security settings.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Nit: point to `remote-audit-settings` ?

**Comment by albertzaharovits** [comment]
I had a single objection about the generic "Generate node certificate" when referring to the transport client comm with a remote cluster. I think this is ambiguous and should be clarified. OTT LGTM

**Review by albertzaharovits — APPROVED** [review]

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc (hunk 4 above)
Comment (lcawl) [reply x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: @albertzaharovits Can you please clarify what change you are suggesting?

Comment (albertzaharovits) [reply x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: @lcawl 
Sorry, I was thinking:
`certs/elastic-certificates.p12` -> `certs/remote-elastic-certificates.p12`

File: x-pack/docs/en/security/auditing/forwarding-logs.asciidoc (hunk 5 above)
Comment (albertzaharovits) [reply x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Following previous comment, my suggestion is:
`/home/es/config/node01.key` -> `/home/es/config/audit-client.key`
`/home/es/config/node01.crt` -> `/home/es/config/audit-client.crt`
`/home/es/config/ca.crt` -> `/home/es/config/remote-ca.crt`

Comment (lcawl) [reply x-pack/docs/en/security/auditing/forwarding-logs.asciidoc]: Thanks @albertzaharovits!  I've made those changes.

**Pull request merged**